src/infra/marketplace/marketplaceCmdRepo.go

```diff
@@ -437,6 +429,139 @@ func (repo *MarketplaceCmdRepo) InstallItem(
+	}
+
+	moveFilesToKeepCmd := fmt.Sprintf(
+		"find %s/ -mindepth 1 -maxdepth 1 -not \\( %s \\) -exec mv -t %s {} +",
+		installedItem.InstallDirectory.String(),
+		fileNameFilterParams,
+		publicDirBackupPath,
+	)
+	_, err = infraHelper.RunCmdWithSubShell(moveFilesToKeepCmd)
+	if err != nil {
+		return errors.New("MoveFilesToKeepToBackupDirectoryError: " + err.Error())
+	}
+
+	srcInstallDirPath := fmt.Sprintf(
+		"/app/%s-%s-%s",
+		installedItem.AppSlug.String(),
+		installedItem.Hostname.String(),
+		installedItem.InstallUuid.String(),
+	)
+	_, err = infraHelper.RunCmdWithSubShell(
+		"mv " + srcInstallDirPath + "/* " + trashDirPath,
+	)
+	if err != nil {
+		return errors.New("MoveFilesToRemoveToTrashError: " + err.Error())
+	}
+
+	_, err = infraHelper.RunCmdWithSubShell(
+		"rm -rf " + srcInstallDirPath,
+	)
+	if err != nil {
+		return errors.New("RemoveSourceInstallDirectoryError: " + err.Error())
+	}
+
+	_, err = infraHelper.RunCmdWithSubShell(
+		"rm -rf " + infraData.GlobalConfigs.PrimaryPublicDir,
+	)
+	if err != nil {
+		return errors.New("RemovePublicDirectoryError: " + err.Error())
+	}
+
+	err = infraHelper.MakeDir(infraData.GlobalConfigs.PrimaryPublicDir)
+	if err != nil {
+		return errors.New("RecreatePublicDirectoryError: " + err.Error())
+	}
+
+	moveFilesToKeepCmd = fmt.Sprintf(
+		"find %s/ -mindepth 1 -maxdepth 1 -exec mv -t %s {} +",
+		publicDirBackupPath,
+		infraData.GlobalConfigs.PrimaryPublicDir,
+	)
+	_, err = infraHelper.RunCmdWithSubShell(moveFilesToKeepCmd)
+	if err != nil {
+		return errors.New("RestoreFilesToKeepError: " + err.Error())
+	}
+
+	_, err = infraHelper.RunCmdWithSubShell("rm -rf " + publicDirBackupPath)
+	if err != nil {
+		return errors.New("RemovePublicDirectoryBackupError: " + err.Error())
+	}
+
+	return nil
+}
+
+func (repo *MarketplaceCmdRepo) uninstallFilesRemoval(
+	installedId valueObject.MarketplaceItemId,
+) error {
+	installedItem, err := repo.marketplaceQueryRepo.ReadInstalledItemById(
+		installedId,
+	)
+	if err != nil {
+		return err
+	}
+
+	catalogItem, err := repo.marketplaceQueryRepo.ReadCatalogItemBySlug(
+		installedItem.AppSlug,
+	)
+	if err != nil {
+		return err
+	}
+
+	if len(catalogItem.UninstallFileNames) == 0 {
+		return nil
+	}
+
+	trashDirPath := fmt.Sprintf(
+		"%s/%s-%s-%s",
+		useCase.TrashDirPath,
+		installedItem.AppSlug.String(),
+		installedItem.Hostname.String(),
+		installedItem.InstallUuid.String(),
+	)
+	err = infraHelper.MakeDir(trashDirPath)
+	if err != nil {
+		return errors.New("CreateTrashDirectoryError: " + err.Error())
+	}
+
+	if infraHelper.IsSymlink(installedItem.InstallDirectory.String()) {
+		return repo.uninstallSymlinkFilesRemoval(installedItem, catalogItem, trashDirPath)
+	}
+
+	fileNameFilterParams := "-name \"" + catalogItem.UninstallFileNames[0].String() + "\""
+	for _, fileToRemove := range catalogItem.UninstallFileNames[1:] {
+		fileNameFilterParams += " -o -name \"" + fileToRemove.String() + "\""
+	}
```

Comment:
I think this happens before. Shouldn't this be an auxiliary method? Maybe the whole idea of mounting the find could be the method so you don't need to mount it twice repeating the code.

Reply:
Done.